clients/mongoStoreClient.go

```diff
@@ -89,19 +89,21 @@ func (d MongoStoreClient) GetTimeLastEntryUser(groupId string) ([]byte, error) {
 	sessionCopy := d.session.Copy()
 	defer sessionCopy.Close()
 
+	// Get the entry with the latest time by reverse sorting and taking the first value
 	err := sessionCopy.DB("").C(DEVICE_DATA_COLLECTION).
 		Find(getBaseQuery(groupId)).
 		Sort("-time").
 		One(&result)
 
-	// Get the entry with the latest time by reverse sorting and taking the first value
-	if err != nil {
+	if err != nil && err != mgo.ErrNotFound {
```

Comment:
First chance I had to look at this.  I know you already merged it, but you might want to change the logic later so it is clearer:
```
if  err != nil {
  if err == go.ErrNotFound {
    // Do something
  } else {
    // Do something else
  }
}
```

I also agree with Jana's comment about refactoring, but that can wait.